From af9421daf405f361fe904f371dac0c00dd1b5ba9 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Thu, 8 Oct 2026 14:35:49 -0600
Subject: [PATCH] fix: Remove the checks for the Python 2 long type in nucname

The functions of nucname.pyx that accept a nuclide as an integer test
for it with

    isinstance(nuc, int) or isinstance(nuc, long)

Cython 3.1 and newer no longer provide the Python 2 name long, so the
file does not compile with them:

    undeclared name not builtin: long

In Python 3 long is the same type as int, which is already tested for,
so the second test is removed.

Refs: https://github.com/cyclus/cyclus Issue 1848
Assisted-by: ClaudeCode:claude-opus-5-5
---
 cyclus/nucname.pyx | 36 ++++++++++++++++++------------------
 1 file changed, 18 insertions(+), 18 deletions(-)

diff --git a/cyclus/nucname.pyx b/cyclus/nucname.pyx
index 579132b9..23bef942 100644
--- a/cyclus/nucname.pyx
+++ b/cyclus/nucname.pyx
@@ -42,7 +42,7 @@ def isnuclide(nuc):
     if isinstance(nuc, basestring):
         nuc_bytes = nuc.encode()
         flag = cpp_nucname.isnuclide(<char *> nuc_bytes)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         flag = cpp_nucname.isnuclide(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -65,7 +65,7 @@ def iselement(nuc):
     if isinstance(nuc, basestring):
         nuc_bytes = nuc.encode()
         flag = cpp_nucname.iselement(<char *> nuc_bytes)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         flag = cpp_nucname.iselement(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -114,7 +114,7 @@ def id(nuc):
     if isinstance(nuc, basestring):
         nuc = nuc.encode()
         newnuc = cpp_nucname.id(<char *> nuc)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         newnuc = cpp_nucname.id(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -165,7 +165,7 @@ def znum(nuc):
     if isinstance(nuc, basestring):
         nuc_bytes = nuc.encode()
         z = cpp_nucname.znum(<char *> nuc_bytes)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         z = cpp_nucname.znum(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -189,7 +189,7 @@ def anum(nuc):
     if isinstance(nuc, basestring):
         nuc_bytes = nuc.encode()
         a = cpp_nucname.anum(<char *> nuc_bytes)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         a = cpp_nucname.anum(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -213,7 +213,7 @@ def snum(nuc):
     if isinstance(nuc, basestring):
         nuc_bytes = nuc.encode()
         s = cpp_nucname.snum(<char *> nuc_bytes)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         s = cpp_nucname.snum(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -237,7 +237,7 @@ def zzaaam(nuc):
     if isinstance(nuc, basestring):
         nuc_bytes = nuc.encode()
         newnuc = cpp_nucname.zzaaam(<char *> nuc_bytes)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         newnuc = cpp_nucname.zzaaam(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -262,7 +262,7 @@ def zzaaam_to_id(nuc):
     if isinstance(nuc, basestring):
         nuc_bytes = nuc.encode()
         newnuc = cpp_nucname.zzaaam_to_id(<char *> nuc_bytes)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         newnuc = cpp_nucname.zzaaam_to_id(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -286,7 +286,7 @@ def zzzaaa(nuc):
     if isinstance(nuc, basestring):
         nuc_bytes = nuc.encode()
         newnuc = cpp_nucname.zzzaaa(<char *> nuc_bytes)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         newnuc = cpp_nucname.zzzaaa(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -310,7 +310,7 @@ def zzzaaa_to_id(nuc):
     """
     if isinstance(nuc, basestring):
         newnuc = cpp_nucname.zzzaaa_to_id(<char *> nuc)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         newnuc = cpp_nucname.zzzaaa_to_id(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -367,7 +367,7 @@ def mcnp_to_id(nuc):
     if isinstance(nuc, basestring):
         nuc_bytes = nuc.encode()
         newnuc = cpp_nucname.mcnp_to_id(<char *> nuc_bytes)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         newnuc = cpp_nucname.mcnp_to_id(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -429,7 +429,7 @@ def zzllaaam(nuc):
     if isinstance(nuc, basestring):
         nuc_bytes = nuc.encode()
         newnuc = cpp_nucname.zzllaaam(<char *> nuc_bytes)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         newnuc = cpp_nucname.zzllaaam(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -605,7 +605,7 @@ def cinder_to_id(nuc):
     """
     if isinstance(nuc, basestring):
         newnuc = cpp_nucname.cinder_to_id(<char *> nuc)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         newnuc = cpp_nucname.cinder_to_id(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -681,7 +681,7 @@ def sza(nuc):
     if isinstance(nuc, basestring):
         nuc_bytes = nuc.encode()
         newnuc = cpp_nucname.sza(<char *> nuc_bytes)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         newnuc = cpp_nucname.sza(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -706,7 +706,7 @@ def sza_to_id(nuc):
     if isinstance(nuc, basestring):
         nuc_bytes = nuc.encode()
         newnuc = cpp_nucname.sza_to_id(<char *> nuc_bytes)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         newnuc = cpp_nucname.sza_to_id(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -730,7 +730,7 @@ def groundstate(nuc):
     if isinstance(nuc, basestring):
         nuc_bytes = nuc.encode()
         newnuc = cpp_nucname.groundstate(<char *> nuc_bytes)
-    elif isinstance(nuc, int) or isinstance(nuc, long):
+    elif isinstance(nuc, int):
         newnuc = cpp_nucname.groundstate(<int> nuc)
     else:
         raise NucTypeError(nuc)
@@ -752,7 +752,7 @@ def state_id_to_id(state):
         Output nuclide in nuc_id form.
 
     """
-    if isinstance(state, int) or isinstance(state, long):
+    if isinstance(state, int):
         newnuc = cpp_nucname.state_id_to_id(<int> state)
     else:
         raise NucTypeError(state)
@@ -774,7 +774,7 @@ def id_to_state_id(nuc):
         Output nuclide in nuc_id form.
 
     """
-    if isinstance(nuc, int) or isinstance(nuc, long):
+    if isinstance(nuc, int):
         newnuc = cpp_nucname.id_to_state_id(<int> nuc)
     else:
         raise NucTypeError(nuc)
-- 
2.56.0

